### PR TITLE
feat(bpdm-cert): initial commit for updated folder structure as per TRG

### LINKED ISSUE #1: Product / repository setup aligned with the TRGs
Hi Sujit,

please take care of the general setup for this specific repository and product. Make sure that all the TRGs are met.
Following the TRG 2 : https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-3

Thanks!

### PR DESCRIPTION
**Pull request by SujitMBRDI**
## Description
Add folder structure and simple api for certificate on following eclipse TRG.

Fixes #1 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: src/main/resources/static/BpdmCertificate.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+openapi: "3.0.2"
```
Comment (carslen) [inline src/main/resources/static/BpdmCertificate.yml:1]: Even to yaml files a copyright and license header should be added, as you did for `application.properties`. 

Please use .yaml as file extension, although .yml isn't wrong and also a well-known extension for YAML files.

File: src/main/resources/static/BpdmCertificate.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    CertificateFullData:
+      allOf:
+        - $ref: '#/components/schemas/Certificate'
+        - type: object
+          properties:
+            certificateId:
+              type: integer
+              description: The ID of the certificate
+              format: int64
+      description: Full data of the certificate.
```
Comment (carslen) [inline src/main/resources/static/BpdmCertificate.yml:67]: Please add new line at EOF.

**Review by carslen — CHANGES_REQUESTED** [review]
Thank you for the new PR! Looks much better now. But I do have some small recommendations, please refer to the comments. After the requested changes have been added to the PR, it is a good starting point for an initial contribution and the IP-Check.

File: src/main/kotlin/org/eclipse/tractusx/bdpmcertificatemanagement/Application.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply src/main/kotlin/org/eclipse/tractusx/bdpmcertificatemanagement/Application.kt:2]: @carslen, thank you for review and for prompt feedback!
I got confused due to example mentioned in TRG [https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02#copyright-and-license-header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02#copyright-and-license-header). I apologize for the same

As you know, this is new product and i am performing initial commits to the project and starting in year 2023.
Could you please confirm first line, will it be
"Copyright (c) 2023,2023 Contributors to the Eclipse Foundation" or
"Copyright (c) 2023 Contributors to the Eclipse Foundation"

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+                            </configOptions>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <plugin>
+                <groupId>org.eclipse.dash</groupId>
+                <artifactId>license-tool-plugin</artifactId>
+                <version>0.0.1-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply pom.xml]: Updated

File: src/main/resources/static/BpdmCertificate.yml (hunk 1 above)
Comment (SujitMBRDI) [reply src/main/resources/static/BpdmCertificate.yml]: Used yaml as file extension

File: src/main/resources/static/BpdmCertificate.yml (hunk 2 above)
Comment (SujitMBRDI) [reply src/main/resources/static/BpdmCertificate.yml]: Done

File: src/main/kotlin/org/eclipse/tractusx/bdpmcertificatemanagement/controller/CertificateController.kt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+@RestController
+@RequestMapping("api/catena/certificate", produces = [MediaType.APPLICATION_JSON_VALUE])
+class CertificateController {
+
+    @GetMapping("/{bpn}")
+    fun getCertificateWelcome(
+        @PathVariable bpn: String
+    ) = "Hello, you requested for certificate $bpn!!"
+}
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply src/main/kotlin/org/eclipse/tractusx/bdpmcertificatemanagement/controller/CertificateController.kt]: Done

File: src/test/kotlin/org/eclipse/tractusx/bdpmcertificatemanagement/ApplicationTests.kt [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package org.eclipse.tractusx.bdpmcertificatemanagement
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply src/test/kotlin/org/eclipse/tractusx/bdpmcertificatemanagement/ApplicationTests.kt:19]: Done

File: SECURITY.md
Comment (carslen) [inline SECURITY.md:1]: Sorry, seems like i missed this in my last review. The content is outdated, please use the content provided in the TRG: https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-01#security-file

**Review by carslen — CHANGES_REQUESTED** [review]

File: SECURITY.md
Comment (SujitMBRDI) [reply SECURITY.md:1]: Adapted new security file.

**Review by carslen — COMMENTED** [review]
Perfect! I'll use this PR for the initial contribution IP-Check at Eclipse-Foundation. After the IP-Check is approved, I'll approve and merge this PR.

**Comment by SujitMBRDI** [comment]
> Created an IP-Check issue at EF: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/11654

Thank you, I can see created IP-Check issue: [https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/11654](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/11654) is approved and closed.

**Review by carslen — APPROVED** [review]
EF IP-Check approved.

**Pull request merged**